source/dem/list_insertion.cc

```diff
@@ -20,27 +45,64 @@ ListInsertion<dim>::ListInsertion(
   // Inializing current inserting particle type
   current_inserting_particle_type = 0;
 
-  const auto &list_x = dem_parameters.insertion_info.list_x;
-  const auto &list_y = dem_parameters.insertion_info.list_y;
-  const auto &list_z = dem_parameters.insertion_info.list_z;
-
+  const auto &list_x  = dem_parameters.insertion_info.list_x;
+  const auto &list_y  = dem_parameters.insertion_info.list_y;
+  const auto &list_z  = dem_parameters.insertion_info.list_z;
+  const auto &list_vx = dem_parameters.insertion_info.list_vx;
+  const auto &list_vy = dem_parameters.insertion_info.list_vy;
+  const auto &list_vz = dem_parameters.insertion_info.list_vz;
+  const auto &list_wx = dem_parameters.insertion_info.list_wx;
+  const auto &list_wy = dem_parameters.insertion_info.list_wy;
+  const auto &list_wz = dem_parameters.insertion_info.list_wz;
+  const auto &list_d  = dem_parameters.insertion_info.list_d;
 
   Assert(list_x.size() == list_y.size(),
-         ListSizeCoherence(list_x.size(), list_y.size()));
+         PositionSizeCoherence(list_x.size(), list_y.size()));
```

Comment:
As before, I would use the maximum vector size here to check for coherence. Maybe you could create a map with list names as keys and sizes as values and then loop assert instead of doing it one by one. I think it would be cleaner this way and you could still print the name (key) of the incoherent list.